bin/compile

```diff
@@ -368,9 +368,9 @@ composer validate --no-plugins --no-check-publish --no-check-all --quiet "$COMPO
 status "Preparing platform package installation..."
 
 if [[ $STACK == "heroku-20" ]]; then
-	HEROKU_PHP_DEFAULT_RUNTIME_VERSION="^7.3.0 | ^8.0.0 <8.4"
+	HEROKU_PHP_DEFAULT_RUNTIME_VERSION="^7.3.0 | ^8.0.0 <8.5"
 else
-	HEROKU_PHP_DEFAULT_RUNTIME_VERSION="^8.0.0 <8.4"
+	HEROKU_PHP_DEFAULT_RUNTIME_VERSION="^8.0.0 <8.5"
```

Comment:
If I'm reading this correctly, apps with an unspecified PHP version can now get PHP 8.4 for the first time. However, this isn't mentioned in the changelog? (The `PHP/8.4.1` entry isn't sufficient to infer that IMO - particularly since  it's a `.1` release, and not `8.4.0`, so doesn't make it clear that this is the first new 8.4 release)

Reply:
It's documented that we always use the latest possible version as the default: https://devcenter.heroku.com/articles/heroku-php-behavior#default-runtime

Reply:
I still think it warrants a mention in the changelog, if the buildpack release has the effect of changing the result of a version resolution to a new major version?

Reply:
If only because of:

> particularly since it's a .1 release, and not 8.4.0, so doesn't make it clear that this is the first new 8.4 release

Reply:
Ah, maybe to clarify - this default only applies to users with basically an empty (or no) `composer.json`. Once you have any requirements for `php` whatsoever in your lock file (and 99.9% of userland libraries specify some kind of version constraint, say `>=7.2.0`), that gets used instead.

The Dev Center changelog entry will mention it, so that's enough IMO (and consistent with previous changelog entries where we did not mention this either).

The constraints there for the default version are more for internal technical reasons anyway (to ensure other branches do not pick up new versions as a default), due to how there is one "global" staging/dev bucket for binaries, rather than something per-branch (moving to which is the longer-term goal).